internal/services/zero_trust_access_service_token/resource.go

```diff
@@ -191,6 +202,14 @@ func (r *ZeroTrustAccessServiceTokenResource) Read(ctx context.Context, req reso
 		resp.Diagnostics.AddError("failed to deserialize http request", err.Error())
 		return
 	}
+
+	if !secret.IsNull() && secret.String() != "" {
+		env.Result.ClientSecret = secret
+		data.ClientSecret = secret
+		resp.State.SetAttribute(ctx, path.Root("client_secret"), &secret)
+		resp.Diagnostics.Append(diags...)
+	}
+
```

Comment:
i think this can be simplified.

```suggestion
	data.ClientSecret = env.Result.ClientSecret
```

Reply:
I think it has to be set to the secret we get from the state? I've simplified the logic. Can you please take a look again?

Reply:
yep, that's correct however, we don't need to `SetAttribute` if we use `data` as a few lines down (L215), it gets passed to `resp.State.Set` anyway and ends up setting it for us.